src/test/java/com/epam/test_generator/services/ProjectServiceTest.java

```diff
@@ -0,0 +1,398 @@
+
+        verify(projectDAO).findOne(eq(simpleProject1.getId()));
+
+    }
+
+    @Test
+    public void getAuthUserFullProject_ProjectIdAndAuth_FullProject() {
+        when(authentication.getPrincipal()).thenReturn(new AuthenticatedUser(
+            null,
+            simpleUser1.getEmail(),
+            null,
+            null,
+            false
+        ));
+        when(projectDAO.findOne(simpleProject1.getId())).thenReturn(simpleProject1);
+        when(userService.getUserByEmail(simpleUser1.getEmail())).thenReturn(simpleUser1);
+        when(projectService.getProjectByProjectId(simpleProject1.getId())).thenReturn(simpleProject1);
+        when(projectFullTransformer.toDto(simpleProject1)).thenReturn(simpleProjectFullDTO1);
+        ProjectFullDTO actualFullProject = projectService
+            .getAuthUserFullProject(simpleProjectFullDTO1.getId(), authentication);
+        assertEquals(simpleProjectFullDTO1, actualFullProject);
+
+        verify(userService).getUserByEmail(eq(simpleUser1.getEmail()));
+        verify(projectFullTransformer).toDto(eq(simpleProject1));
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void getAuthUserFullProject_UserDoesntBelongsToSuit_BadRequestException() {
+        simpleProject1.setUsers(Stream.of(simpleUser2).collect(Collectors.toSet()));
+        when(authentication.getPrincipal()).thenReturn(new AuthenticatedUser(
+            null,
+            simpleUser1.getEmail(),
+            null,
+            null,
+            false
+        ));
+        when(projectDAO.findOne(simpleProject1.getId())).thenReturn(simpleProject1);
+        when(userService.getUserByEmail(simpleUser1.getEmail())).thenReturn(simpleUser1);
+        when(projectService.getProjectByProjectId(simpleProject1.getId())).thenReturn(simpleProject1);
+        ProjectFullDTO actualFullProject = projectService
+            .getAuthUserFullProject(simpleProjectFullDTO1.getId(), authentication);
+    }
+
+    @Test
+    public void createProject() {
+    }
+
+    @Test
+    public void updateProject() {
+    }
+
+    @Test
+    public void removeProject_valid_Success() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        projectService.removeProject(SIMPLE_PROJECT_ID);
+        verify(projectDAO).delete(SIMPLE_PROJECT_ID);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void removeProject_invalidProjectId_NotFound() {
+
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        projectService.removeProject(777L);
+        verify(projectDAO).delete(SIMPLE_PROJECT_ID);
+    }
+
+    @Test
+    public void addUserToProject_valid_Success() {
+        simpleProject1.setUsers(new HashSet<>());
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        assertFalse(simpleProject1.getUsers().contains(simpleUser1));
```

Comment:
this assertion is pointless. It does not validate any piece of production code.

Reply:
we use it to prove that simpleProject1 doesnt contain any users (especially simpleUser1) before 
#addUserToProject invokes. (more explonation in comment bellow)

Reply:
or did you mean that we dont need  #assertFalse? (i thougth you mean whole test method is general)